File: CFNDeploy/csharp/src/PipelineStack.cs
using Amazon.CDK;
using Amazon.CDK.Pipelines;
using Constructs;

/// <summary>
/// Pipeline stack with multi-stage deployment.
/// </summary>
namespace CFNDeploy
{
    public class PipelineStack : Stack
    {
        internal PipelineStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            var pipeline = new CodePipeline(this, "Pipeline", new CodePipelineProps
            {
                PipelineName = "CFNDeployPipeline",
                Synth = new ShellStep("Synth", new ShellStepProps
                {
                    Input = CodePipelineSource.GitHub("aws-samples/aws-cdk-examples", "main"),
                    Commands = new[] { "dotnet build src", "npx cdk synth" }
                })
            });

            var testStage = pipeline.AddStage(new AppStage(this, "Test", new StageProps
            {
                Env = new Amazon.CDK.Environment
                {
                    Account = System.Environment.GetEnvironmentVariable("CDK_DEFAULT_ACCOUNT"),
                    Region = System.Environment.GetEnvironmentVariable("CDK_DEFAULT_REGION")
                }
            }));

            testStage.AddPre(new ShellStep("UnitTest", new ShellStepProps
            {
                Commands = new[] { "dotnet test src" }
            }));

            var prodStage = pipeline.AddStage(new AppStage(this, "Prod", new StageProps
            {
                Env = new Amazon.CDK.Environment
                {
                    Account = System.Environment.GetEnvironmentVariable("CDK_DEFAULT_ACCOUNT"),
                    Region = System.Environment.GetEnvironmentVariable("CDK_DEFAULT_REGION")
                }
            }));

            prodStage.AddPre(new ManualApprovalStep("PromoteToProd"));
        }
    }
}

File: StepFunctionInvoke/java/src/main/java/com/example/PipelineStack.java
package com.example;

/**
 * Pipeline stack with multi-stage deployment.
 */

import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.pipelines.CodePipeline;
import software.amazon.awscdk.pipelines.CodePipelineSource;
import software.amazon.awscdk.pipelines.ShellStep;
import software.amazon.awscdk.pipelines.ManualApprovalStep;
import software.constructs.Construct;
import java.util.Arrays;

public class PipelineStack extends Stack {
    public PipelineStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        CodePipeline pipeline = CodePipeline.Builder.create(this, "Pipeline")
                .pipelineName("StepFunctionInvokePipeline")
                .synth(ShellStep.Builder.create("Synth")
                        .input(CodePipelineSource.gitHub("aws-samples/aws-cdk-examples", "main"))
                        .commands(Arrays.asList("mvn compile", "npx cdk synth"))
                        .build())
                .build();

        AppStage testStage = new AppStage(this, "Test", software.amazon.awscdk.StageProps.builder()
                .env(software.amazon.awscdk.Environment.builder()
                        .account(System.getenv("CDK_DEFAULT_ACCOUNT"))
                        .region(System.getenv("CDK_DEFAULT_REGION"))
                        .build())
                .build());

        pipeline.addStage(testStage).addPre(ShellStep.Builder.create("UnitTest")
                .commands(Arrays.asList("mvn test"))
                .build());

        AppStage prodStage = new AppStage(this, "Prod", software.amazon.awscdk.StageProps.builder()
                .env(software.amazon.awscdk.Environment.builder()
                        .account(System.getenv("CDK_DEFAULT_ACCOUNT"))
                        .region(System.getenv("CDK_DEFAULT_REGION"))
                        .build())
                .build());

        pipeline.addStage(prodStage).addPre(new ManualApprovalStep("PromoteToProd"));
    }
}
